### PR TITLE
Feature eks

### PR DESCRIPTION
**Pull request by tprakash17**
Terraform template for eks cluster. Has been tested within AWS test account.

### CODE REVIEW

File: cluster-addons/cluster-autoscalar/ca-iam-policy.tf [hunk 1]
```diff
@@ -0,0 +1,59 @@
+resource "aws_iam_role_policy" "frankfurt_ca_policy" {
+  name = "frankfurt_ca_policy"
```
Comment (ntantri) [inline cluster-addons/cluster-autoscalar/ca-iam-policy.tf]: Can we not name it `frankfurt`? It should be a generic one.

File: aws-vpn-gtw.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# create aws vpn gateway for EKS VPC Frankfurt
+resource "aws_vpn_gateway" "vpn_gw" {
+  vpc_id = "${aws_vpc.frankfurt.id}"
+
+  tags = "${
+    map(
+     "Name", "eks aws vpn gateway frankfurt"
```
Comment (ntantri) [inline aws-vpn-gtw.tf]: Can we take all the regions as a parameter? Not hardcode with Frankfurt or eu-east-1 and so on.

File: backend.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+## A "backend" in Terraform determines how state is loaded. Its completely optional but recommended.
+## Terraform remote state management - visit https://www.terraform.io/docs/backends/index.html
+## eks-frankfurt is the folder inside the bucket that you are going to choose to store terraform state files. 
+## make sure you create it in advance.
+
+#terraform {
+#  backend "s3" {
+#    bucket = "{var.s3_bucket_name}"
+#    key    = "eks-frankfurt/terraform.tfstate"
+#    region = "us-east-1"
```
Comment (ntantri) [inline backend.tf]: Here you say `eks-frankfurt` with region as `us-east-1` which can be confusing.

File: cluster-addons/cluster-autoscalar/eks-ca.yaml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      containers:
+        - image: k8s.gcr.io/cluster-autoscaler:v1.3.0
+          name: cluster-autoscaler
+          resources:
+            limits:
+              cpu: 100m
+              memory: 300Mi
+            requests:
+              cpu: 100m
+              memory: 300Mi
```
Comment (ntantri) [inline cluster-addons/cluster-autoscalar/eks-ca.yaml]: Can we ensure these limits are passed via command line?

File: cluster-addons/ingress-controller/private-ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+    # replace with the correct value of the generated certificate in the AWS console Mediaiqdigital.com
+    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: "arn:aws:acm:eu-central-1:230367374156:certificate/d5ea0711-b37c-49de-a407-f5857f09d229"
+    # the backend instances are HTTP
+    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: "http"
+    # Map port 443
+    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "https"
+    # Increase the ELB idle timeout to avoid issues with WebSockets or Server-Sent Events.
+    service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout: '3600'
+    # tags for ELB
+    service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags: "Name=frankfurt-private-ingess-controller,TEAM=Devops,PRODUCT=EKS,ENVIRONMENT=PRODUCTION"
```
Comment (ntantri) [inline cluster-addons/ingress-controller/private-ingress.yaml]: the name has again got Frankfurt added in. Let's not mention anything of the region in these files.

File: eks-cluster.tf [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+resource "aws_iam_role_policy_attachment" "frankfurt-cluster-AmazonEKSClusterPolicy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"
+  role       = "${aws_iam_role.frankfurt-cluster.name}"
+}
+
+resource "aws_iam_role_policy_attachment" "frankfurt-cluster-AmazonEKSServicePolicy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSServicePolicy"
+  role       = "${aws_iam_role.frankfurt-cluster.name}"
+}
```
Comment (ntantri) [inline eks-cluster.tf:35]: Let's not mention frankfurt

**Review by ntantri — CHANGES_REQUESTED** [review]
A couple of things we need to change, naming and so on.

File: cluster-addons/cluster-autoscalar/eks-ca.yaml (hunk 4 above)
Comment (tprakash17) [reply cluster-addons/cluster-autoscalar/eks-ca.yaml]: removed add ons - not required

File: cluster-addons/cluster-autoscalar/ca-iam-policy.tf (hunk 1 above)
Comment (tprakash17) [reply cluster-addons/cluster-autoscalar/ca-iam-policy.tf]: done

File: eks-cluster.tf (hunk 6 above)
Comment (tprakash17) [reply eks-cluster.tf:35]: done

**Pull request closed without merging**